LoopKit/GlucoseRangeSchedule.swift

```diff
@@ -163,31 +170,30 @@ public struct GlucoseRangeSchedule: DailySchedule {
         self.overrideRanges = overrideRanges
 
         if let overrideRawValue = rawValue["override"] as? Override.RawValue {
-            self.override = Override(rawValue: overrideRawValue)
+            self.override = Override(rawValue: overrideRawValue, unit: rangeSchedule.unit)
         }
     }
 
     public func between(start startDate: Date, end endDate: Date) -> [AbsoluteScheduleValue<DoubleRange>] {
         return rangeSchedule.between(start: startDate, end: endDate)
     }
 
-    public func quantityBetween(start: Date, end: Date) -> [AbsoluteScheduleValue<Range<HKQuantity>>] {
-        var quantitySchedule = [AbsoluteScheduleValue<Range<HKQuantity>>]()
+    public func quantityBetween(start: Date, end: Date) -> [AbsoluteScheduleValue<ClosedRange<HKQuantity>>] {
+        var quantitySchedule = [AbsoluteScheduleValue<ClosedRange<HKQuantity>>]()
 
         for schedule in between(start: start, end: end) {
-            let lowerBound = HKQuantity(unit: unit, doubleValue: schedule.value.minValue)
-            let upperBound = HKQuantity(unit: unit, doubleValue: schedule.value.maxValue)
-
             quantitySchedule.append(AbsoluteScheduleValue(
                 startDate: schedule.startDate,
                 endDate: schedule.endDate,
-                value: lowerBound..<upperBound
+                value: schedule.value.quantityRange(for: unit)
             ))
         }
 
         return quantitySchedule
     }
 
+    /// Returns the underlying values in `unit`
+    /// Consider using quantity(at:) instead
```

Comment:
Is this a TODO? Under what considerations should we use `quantity(at:)`? 

Also, is it `quantityRange(at:)` instead?

Reply:
In general it’s better to use unit-included values wherever possible, especially with glucose because it’s a preference that a user can change at any time. So the comment means “if you’re thinking about using this value(at:) method, consider using quantity(at:) instead”

Reply:
In general it’s better to use unit-included values wherever possible, especially with glucose because it’s a preference that a user can change at any time. So the comment means “if you’re thinking about using this value(at:) method, consider using quantity(at:) instead”